### PR TITLE
Cache vector in ObjectiveArrow.lua

### PR DESCRIPTION
**Pull request by 4z0t**

### CODE REVIEW

**Comment by Hdt80bro** [comment]
@Garanas is changing the signature of `BounceThread` to accept the saved offset and vector safe to do so that it doesn't have to go through instance fields?

**Comment by Hdt80bro** [comment]
Actually, we could move `BounceTime` from the instance as well if it's not used anywhere else (which it doesn't look like it is, but, well, mods are weird sometimes).

**Comment by 4z0t** [comment]
I can make this vector a local variable as well as offset

File: lua/objectiveArrow.lua [hunk 1]
```diff
@@ -19,25 +19,26 @@ ObjectiveArrow = Class(Entity) {
...
                 yOff = (self.Size * unitScale) / 2.0 + extents.Max.y - spec.AttachTo:GetPosition().y
                 yOff = yOff + 0.5
             else
                 yOff = spec.Size / 2.0 + 0.5
             end
 
-            self:SetParentOffset(Vector(0,yOff,0))
             self.SavedOffset = yOff;
-            ForkThread(self.BounceThread,self)
+            self.SavedVector = Vector(0, yOff, 0)
```
Comment (Garanas) [inline lua/objectiveArrow.lua:39]: Probably best to just pass the offset to BounceThread, and allocate a vector in the local scope of the thread. That way there is no 'global' scope that can mess things up.

Comment (Garanas) [reply lua/objectiveArrow.lua:39]: @Hdt80bro ^, I think that matches your suggestion

**Review by Hdt80bro — APPROVED** [review]

**Comment by Hdt80bro** [comment]
I had been staring at this too long, I needed to put my changes somewhere so I could move on. I managed to exclude all optimizations I had stored that would have reduced readability such as Chebyshev recursion and variable reusing though.

**Comment by 4z0t** [comment]
@Hdt80bro I think this is redundant at this point

**Comment by Hdt80bro** [comment]
What else are you thinking covers these changes?

**Pull request merged**